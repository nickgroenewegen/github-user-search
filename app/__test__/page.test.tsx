import { useFormState } from 'react-dom'
import { render, screen } from '@testing-library/react'
import Page from '../page'

jest.mock('react-dom', () => ({
  __esModule: true,
  ...jest.requireActual('react-dom'),
  useFormState: jest.fn(() => []),
}))

test('Correctly renders the home and should be stable', () => {
  const { container } = render(<Page />)

  const heading = screen.getByRole('heading', { level: 1 })
  const paragraph = screen.getByRole('paragraph')
  const form = screen.getByRole('search')

  expect(heading).toBeInTheDocument()
  expect(paragraph).toBeInTheDocument()
  expect(form).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
