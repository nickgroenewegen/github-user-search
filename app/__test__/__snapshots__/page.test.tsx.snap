// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the home and should be stable 1`] = `
<div>
  <div
    class="flex"
  >
    <div>
      <h1>
        GitHub user search
      </h1>
      <p>
        Query the GitHub search API to find users or organizations you're looking for.
      </p>
    </div>
    <form
      class="search-form"
      data-test-id="search-form"
      role="search"
    >
      <label
        class="sr-only"
        for="search"
      >
        Search term
      </label>
      <input
        autocomplete="off"
        id="search"
        minlength="3"
        name="q"
        placeholder="e.g. john, some@email.com"
        required=""
        type="search"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
